seqr/views/apis/staff_api_tests.py

```diff
@@ -0,0 +1,173 @@
+          "genomeVersion": "38",
+          "sampleType_1": "WGS",
+          "sourceFilePath_1": "gs://seqr-datasets/v02/GRCh38/RDG_WGS_Broad_Internal/v10/sharded_vcf/RDG_Broad_WGS_Internal.filtered.*.vcf.gz",
+          "genomeVersion_1": "38",
+          "sampleType": "WGS",
+          "sourceFilePath": TEST_INDEX_SECOND_SRC_PATH,
+          "hail_version_1": "0.2.24",
+          "gencodeVersion_1": "25"
+        },
+        "_all": {
+          "enabled": False
+        },
+      }
+    }
+  },
+  "test_index_old": {
+    "mappings": {
+      "variant": {
+        "_meta": {
+          "gencodeVersion": "25",
+          "hail_version": "0.2.24",
+          "genomeVersion": "38",
+          "sampleType_1": "WGS",
+          "sourceFilePath_1": "gs://seqr-datasets/v02/GRCh38/RDG_WGS_Broad_Internal/v10/sharded_vcf/RDG_Broad_WGS_Internal.filtered.*.vcf.gz",
+          "genomeVersion_1": "38",
+          "sampleType": "WGS",
+          "sourceFilePath": TEST_INDEX_SRC_PATH,
+          "hail_version_1": "0.2.24",
+          "gencodeVersion_1": "25"
+        },
+        "_all": {
+          "enabled": False
+        },
+      }
+    }
+  },
+  "1kg.vcf.gz": {
+    "mappings": {
+      "variant": {
+        "_meta": {
+          "gencodeVersion": "19",
+          "genomeVersion": "37",
+          "sampleType": "WES",
+          "datasetType": "VARIANTS",
+          "sourceFilePath": TEST_INDEX_1KG_SRC_PATH
+        },
+        "_all": {
+          "enabled": False
+        },
+      }
+    }
+  }
+}
+from django.test import TestCase
+from django.urls.base import reverse
+
+from seqr.views.apis.staff_api import elasticsearch_status
+from seqr.views.utils.test_utils import _check_login
+
+class VariantSearchAPITest(TestCase):
+    fixtures = ['users', '1kg_project']
+    multi_db = True
+
+    @mock.patch('elasticsearch_dsl.index.Index.get_mapping')
+    @mock.patch('elasticsearch.client.cat.CatClient.aliases')
+    @mock.patch('elasticsearch.client.cat.CatClient.indices')
+    @mock.patch('elasticsearch.client.cat.CatClient.allocation')
+    def test_elasticsearch_status(self, mock_cat_allocation, mock_cat_indices, mock_cat_aliases, mock_get_mapping):
+        url = reverse(elasticsearch_status)
+        _check_login(self, url)
+
+        mock_cat_allocation.return_value = ES_CAT_ALLOCATION
+        mock_cat_indices.return_value = ES_CAT_INDICES
+        mock_cat_aliases.return_value = ES_CAT_ALIAS
+        mock_get_mapping.return_value = ES_INDEX_MAPPING
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['indices', 'errors', 'diskStats', 'elasticsearchHost'])
+        for index in response_json['indices']:
+            index_id = index['index']
+            mapping = ES_INDEX_MAPPING[index_id]
+            self.assertEqual(index['sourceFilePath'], mapping['mappings']['variant']['_meta']['sourceFilePath'])
```

Comment:
rather than testing that one field of every dictionary has the correct value, you should only have 2 dictionaries (maybe one that has an active project and one that doesn't) and check that all the dictionary values are correct using `assertDictEquals`

Reply:
Good idea. I didn't know it can check the dict as a whole.

Reply:
It has been found out that the dicts are not the same type. So, it can't use "assertDictEquals". I have added all the meta fields and project item checking.

Reply:
I removed index `test_index_old` from the elasticsearch response to generate an `errors` output. And I found that there is a bug in the api elasticsearch_status() in below lines: (The `p.name` must be `p`)
```
errors = ['{} does not exist and is used by project(s) {}'.format(
        index, ', '.join(['{} ({} samples)'.format(p.name, len(indivs)) for p, indivs in project_individuals.items()])
    ) for index, project_individuals in seqr_index_projects.items() if project_individuals]
```

Reply:
You should be able to deterministically know exactly what the content of the response dictionary is, and you should therefore be checking the entirety of the content. Iterating and using variables like this make it really hard to tell what the expected data looks like, which makes understanding future changes really difficult. What does it mean for the dicts to not have the same type? You are the one writing the dict it is being compared to, so you can make it the same

Reply:
Okay, I will make expected result dicts and compare them again the test results.